src/gwt/src/org/rstudio/studio/client/common/rpubs/ui/RPubsUploadDialog.java

```diff
@@ -104,25 +105,25 @@ protected Widget createMainWidget()
       String msg;
       if (!isPublished_ && uploadId_.isEmpty())
       {
-         msg = "RPubs is a free service from RStudio for sharing " +
-                       "documents on the web. Click Publish to get " +
-                       "started.";
+         msg = constants_.rPubsServiceMessage() +
+                       constants_.clickPublishMessage() +
+                       constants_.startedMessage();
       }
       else
       {
-         msg = "This document has already been published on RPubs. You can " +
-               "choose to either update the existing RPubs document, or " +
-               "create a new one.";
+         msg = constants_.alreadyPublishedRPubs() +
+               constants_.updateRPubsMessage() +
+               constants_.createNewRPub();
       }
       Label descLabel = new Label(msg);
       descLabel.addStyleName(styles.descLabel());
       verticalPanel.add(descLabel);
       setARIADescribedBy(descLabel.getElement());
 
       HTML warningLabel =  new HTML(
-        "<strong>IMPORTANT: All documents published to RPubs are " +
-        "publicly visible.</strong> You should " +
-        "only publish documents that you wish to share publicly.");
+        "<strong>" + constants_.importantRPubsMessage() +
+         constants_.publiclyVisibleWarningLabel() + "</strong> " + constants_.youShouldMessage() +
+        constants_.publishDocumentsMessage());
       warningLabel.addStyleName(styles.warningLabel());
       verticalPanel.add(warningLabel);
```

Comment:
Use a string with placeholders, not concat, so probably:

```
"<strong>{0}</strong> {1}"
```

Reply:
Can this be resolved?